src/main/java/at/codersbay/datastructures/Fractional.java

```diff
@@ -5,70 +5,133 @@
  * e.g.: 1/2, 1/3, 24/37, ..
  */
 public class Fractional {
+	
+	private Integer numerator;
+	private Integer denominator;
 
     public Fractional(Integer numerator, Integer denominator) {
-        // TODO implement to fix tests in FractionalTest1
+    	/*numerator can be 0 but not null*/
+    	if(numerator == null) {
+    		throw new IllegalArgumentException();
+    	} else {
+    		this.numerator = numerator;
+    	}
+    	
+    	/*denominator cant be 0 or null*/
+    	if(denominator == null || denominator == 0) {
+    		throw new IllegalArgumentException();
+    	} else {
+    		this.denominator = denominator;
+    	}
+        
+    }
+    
+    public Fractional invertedFractional(Fractional fractional) {
+    	Fractional returnFractional = new Fractional(
+    		fractional.getDenominator(), fractional.getNumerator());
+    	return returnFractional;	
     }
 
     public Integer getNumerator() {
-        // TODO implement to fix tests in FractionalTest1
-        return -1;
+        return this.numerator;
     }
 
     public Integer getDenominator() {
-        // TODO implement to fix tests in FractionalTest1
-        return -1;
+        return this.denominator;
     }
 
     public Float asFloat() {
-        // TODO implement to fix tests in FractionalTest2
-        return -1f;
+        return  ((float)this.numerator/(float) this.denominator);
     }
 
-    @Override
-    public boolean equals(Object obj) {
-        // TODO implement to fix tests in FractionalTest3
-        return false;
-    }
+	@Override
+	public boolean equals(Object obj) {
+		Fractional fractionalObject;
+		/*!Obj is not a Fractional or is null*/
+		if (!obj.getClass().equals(this.getClass()) || obj == null) {
```

Comment:
I would rather return false here instead of throwing an IAE - but that's a design decision

Reply:
aye